src/logger/mod.rs

```
@@ -0,0 +1,85 @@
+    extra: Option<String>,
+}
+
+impl Log {
+    fn new(log_type: LogType, extra: Option<String>) -> Self {
+        let event = match log_type {
+            LogType::AddPassword => "add_password".to_string(),
+            LogType::List => "list".to_string(),
+            LogType::RemovePassword => "remove_password".to_string(),
+            LogType::GeneratePassword => "generate_password".to_string(),
+            LogType::Show => "show".to_string(),
+            LogType::UpdateMasterPassword => "update_master".to_string(),
+        };
+        let time_stamp = Local::now().to_string();
+        Self {
+            event,
+            time_stamp,
+            extra,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct LogsData {
+    logs: Vec<Log>,
+}
+
+pub struct Logger {
+    path: PathBuf,
+}
+
+impl Logger {
+    pub fn new() -> Self {
+        let file_name = "logs.json".to_string();
+        let path = get_password_store_path(file_name).unwrap();
+        let log_file = File::open(&path);
+        match log_file {
+            Err(e) => match e.kind() {
+                ErrorKind::NotFound => {
+                    let john = json!({
+                        "logs": []
+                    });
+                    let mut file = File::create(&path).unwrap();
+                    file.write_all(john.to_string().as_bytes()).unwrap();
+                }
+                _ => (),
+            },
+            _ => (),
+        }
+        Self { path }
+    }
+
+    // Write log to the file
+    pub fn write_log(&self, log_type: LogType, extra: Option<String>) {
+        let file_stringified = read_to_string(&self.path).unwrap();
+        let mut json_data: LogsData = serde_json::from_str(&file_stringified).unwrap();
+        let log = Log::new(log_type, extra);
+        json_data.logs.push(log);
+        let json_value = serde_json::to_string_pretty(&json_data).unwrap();
+        std::fs::write(&self.path, json_value).unwrap();
```

It'd be good if we can handle this error more gracefully, rather than using `unwarp`. We can propagate the error to the caller, and the caller can print a warning message like "Failed to log: {err_message}`.